crates/pathfinder/src/rpc/v02/method/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::Context;
+use serde_with::serde_as;
+
+use crate::rpc::serde::{FeeAsHexStr, TransactionVersionAsHexStr};
+use crate::{
+    core::{
+        CallParam, ClassHash, ContractAddress, ContractAddressSalt, Fee, StarknetTransactionHash,
+        TransactionNonce, TransactionSignatureElem, TransactionVersion,
+    },
+    rpc::v02::RpcContext,
+    sequencer::ClientApi,
+};
+
+const fn transaction_version_zero() -> TransactionVersion {
+    TransactionVersion(web3::types::H256::zero())
+}
+
+#[serde_as]
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub struct AddDeployAccountTransactionInput {
+    // Fields from BROADCASTED_TXN_COMMON_PROPERTIES
+    #[serde_as(as = "TransactionVersionAsHexStr")]
+    #[serde(default = "transaction_version_zero")]
+    pub version: TransactionVersion,
+    #[serde_as(as = "FeeAsHexStr")]
+    pub max_fee: Fee,
+    pub signature: Vec<TransactionSignatureElem>,
+    pub nonce: TransactionNonce,
+
+    // Fields from DEPLOY_ACCOUNT_TXN_PROPERTIES
+    pub contract_address_salt: ContractAddressSalt,
+    pub constructor_calldata: Vec<CallParam>,
+    pub class_hash: ClassHash,
+}
+
+#[derive(serde::Serialize, Debug, PartialEq, Eq)]
+pub struct AddDeployAccountTransactionOutput {
+    transaction_hash: StarknetTransactionHash,
+    contract_address: ContractAddress,
+}
+
+crate::rpc::error::generate_rpc_error_subset!(AddDeployAccountTransactionError: ClassHashNotFound);
+
+pub async fn add_deploy_account_transaction(
+    context: RpcContext,
+    input: AddDeployAccountTransactionInput,
+) -> Result<AddDeployAccountTransactionOutput, AddDeployAccountTransactionError> {
+    let response = context
+        .sequencer
+        .add_deploy_account(
+            input.version,
+            input.max_fee,
+            input.signature,
+            input.nonce,
+            input.contract_address_salt,
+            input.class_hash,
+            input.constructor_calldata,
+        )
+        .await
+        .context("Senging Deplpy Account Transaction to the gateway")?;
```

Comment:
Just some minor typos:
```suggestion
        .context("Sending Deploy Account Transaction to the gateway")?;
```

Reply:
Thanks, good catch.